internal/provider/resource_subaccount_subscription.go

```diff
@@ -248,6 +248,10 @@ func (rs *subaccountSubscriptionResource) Create(ctx context.Context, req resour
 		resp.Diagnostics.AddError("API Error Creating Resource Subscription (Subaccount)", fmt.Sprintf("%s", err))
 	}
 
+	if updatedRes == nil {
```

Comment:
I think this could be directly added to the statement above: 

```go
if err != nil {
		resp.Diagnostics.AddError("API Error Creating Resource Subscription (Subaccount)", fmt.Sprintf("%s", err))
return
	}
```

Or do we have a situation where no error is returned but the updatedRes is `nil`?

Reply:
There may be scenarios where a subscription gets created in the FAILED state. In such cases a response is returned with the ```state``` parameter populated as ```FAILED``` which is how it is stored as a tainted resource in the state file. If the ```return``` statement is placed within the ```if block```, then such a resource isn't maintained in the state file although the subscription is created on the platform.

<img width="597" alt="image" src="https://github.com/user-attachments/assets/4b12ec90-0369-4159-acc2-bf5222f5ceda">

<img width="1411" alt="image" src="https://github.com/user-attachments/assets/27ed967f-5ea0-4143-9c3f-de1203260a06">